blackjack/lib/blackjack/game.ex

```diff
@@ -0,0 +1,162 @@
+defmodule Blackjack.Game do
+
+  # New game method
+  def new_game() do
+    %Blackjack.State{}
+      |> Map.put(:deck_state, Deck.shuffle())
+  end
+
+  # State  
+  def check(game) do
+  %{
+    game_state: game.game_state,
+    deck_state: game.deck_state,
+    dealer_hand: game.dealer_hand,
+    dealer_value: game.dealer_value,
+    player_hand: game.player_hand,
+    player_value: game.player_value,
+    turn: game.turn
+  }
+  end
+
+  # New game setup
+  def hit(game = %{ game_state: :init }) do
+    {[card1, card2, card3, card4], new_deck} = Deck.deal(game.deck_state, 4)
```

Comment:
I think the following might be a good time to use:

~~~ elixir
 %BlackJack.Game{ game | game_state: ongoing, player_hand: [ card1, card3 ], ...}
~~~